udp-echo/src/main.rs

```diff
@@ -1,42 +1,58 @@
-use actix::{Actor, AsyncContext, Context, Message, StreamHandler};
+use actix::io::SinkWrite;
+use actix::{Actor, Arbiter, AsyncContext, Context, Message, StreamHandler};
+use bytes::Bytes;
 use bytes::BytesMut;
 use futures::stream::SplitSink;
-use futures::{Future, Sink, Stream};
+use futures_util::stream::StreamExt;
+use std::io::Result;
 use std::net::SocketAddr;
-use tokio::codec::BytesCodec;
-use tokio::net::{UdpFramed, UdpSocket};
+use tokio::net::UdpSocket;
+use tokio_util::codec::BytesCodec;
+use tokio_util::udp::UdpFramed;
+
+type SinkItem = (Bytes, SocketAddr);
+type UdpSink = SplitSink<UdpFramed<BytesCodec>, SinkItem>;
 
 struct UdpActor {
-    sink: SplitSink<UdpFramed<BytesCodec>>,
+    sink: SinkWrite<SinkItem, UdpSink>,
 }
 impl Actor for UdpActor {
     type Context = Context<Self>;
 }
 
 #[derive(Message)]
+#[rtype(result = "()")]
 struct UdpPacket(BytesMut, SocketAddr);
-impl StreamHandler<UdpPacket, std::io::Error> for UdpActor {
+
+impl StreamHandler<UdpPacket> for UdpActor {
     fn handle(&mut self, msg: UdpPacket, _: &mut Context<Self>) {
         println!("Received: ({:?}, {:?})", msg.0, msg.1);
-        (&mut self.sink).send((msg.0.into(), msg.1)).wait().unwrap();
+
+        self.sink.write(("PING\n".into(), msg.1)).unwrap();
     }
 }
 
+impl actix::io::WriteHandler<std::io::Error> for UdpActor {}
+
 fn main() {
     let sys = actix::System::new("echo-udp");
 
-    let addr: SocketAddr = "127.0.0.1:0".parse().unwrap();
-    let sock = UdpSocket::bind(&addr).unwrap();
-    println!(
-        "Started udp server on: 127.0.0.1:{:?}",
-        sock.local_addr().unwrap().port()
-    );
```

Comment:
I think the start message here should also be preserved.

Reply:
Fixed.